### PR TITLE
add: debian packaging using goreleaser

### PR DESCRIPTION
**Pull request by malaupa**
* goreleaser add to build deb package
* github workflows added for ci and release flow

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
 # Output of the go coverage tool, specifically when used with LiteIDE
 *.out
-
+# Test binary, build with `go test -c`
+*.test
+# Ignore dist
+dist/
 # Ignore executables
-/fw-id-agent
-/fw-id-cli
```
Comment (hwipl) [inline .gitignore:16]: Can we keep the "/" before the executables?

File: .goreleaser.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      - "^docs:"
+      - "^test:"
+nfpms:
+  - package_name: fw-id-agent
+    file_name_template: "{{ .ProjectName }}_{{ .Version }}_{{ .Arch }}"
+    vendor: Deutsche Telekom MMS GmbH
+    maintainer: The MMS Linux Dev Team <mmslinux-dev@t-systems-mms.com>
+    description: |-
+      Corporate Identity Agent
+      Agent to logon at the corporate firewall.
```
Comment (hwipl) [inline .goreleaser.yml:60]: I would prefer the name "Firewall Identity Agent" so it matches the title we have in the README.
Some alternative suggestions for the description: "Agent to log in on the corporate firewall.", "Agent for logging in on the corporate firewall." "...log on to...", "... logging on to..."

File: go.mod [hunk 3]
```diff
@@ -4,14 +4,14 @@ go 1.18
 
 require (
 	github.com/T-Systems-MMS/tnd v0.0.0-20230404085219-434ecc2a5379
+	github.com/fsnotify/fsnotify v1.6.0
```
Comment (hwipl) [inline go.mod:7]: nit-pick: this could be moved to a new PR with the other cleanups that were previously in this PR.

File: .goreleaser.yml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      preremove: build/package/preremove.sh
+      postremove: build/package/postremove.sh
+    deb:
+      lintian_overrides:
+        - "no-manual-page"
+        - "no-changelog"
+        - "statically-linked-binary"
+        - "maintainer-script-calls-systemctl"
+    dependencies:
+      - "libc6 (>= 2.34)"
```
Comment (hwipl) [inline .goreleaser.yml:88]: Not sure we need this with the statically linked executables. Is this a debian packaging thing?

File: LICENSE [hunk 5]
```diff
@@ -1,14 +1,21 @@
-Copyright 2022 T-Systems Multimedia Solutions GmbH
+MIT License
 
-Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
-documentation files (the "Software"), to deal in the Software without restriction, including without limitation
-the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
-and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+Copyright (c) 2023 Deutsche Telekom MMS GmbH
```
Comment (jandd) [inline LICENSE:3]: Copyright (c) 2022-2023 Deutsche Telekom MMS GmbH

or without a year for easier maintenance

File: .goreleaser.yml (hunk 4 above)
Comment (malaupa) [reply .goreleaser.yml]: I removed it after i checked the binary with ldd.

**Pull request merged**